BlueDB/src/main/java/org/bluedb/api/encryption/EncryptionConfig.java

```diff
@@ -0,0 +1,9 @@
+package org.bluedb.api.encryption;
+
+public interface EncryptionConfig {
```

Comment:
I'm okay with having one interface instead of a read and read-write one and wrapping it. However, I don't think `EncryptionConfig` is quite right for the name. It isn't just a model contain configuration information. The client will be implementing it with functionality that actually does the encryption and decryption. So I feel like it is much more that a config object. This is also the publicly visible object, so I'd rather have it have a good name than our wrapper. How do you feel about naming this class `EncryptionService` and naming the wrapper `EncryptionServiceWrapper` or something like that? Or any other ideas?

Reply:
I agree `EncryptionConfig` isn't a great name, I was trying to encapsulate the idea that this is a user supplied configurable object. I have been racking my brain for a better name than what you suggested but I think that is pretty accurate to what we are doing, I'll go ahead and make that change!